src/monocle_apptrace/wrap_common.py

```diff
@@ -291,10 +291,11 @@ def update_llm_endpoint(curr_span: Span, instance):
 
 def set_provider_name(instance):
     provider_url = ""
-
+    inference_endpoint = ""
     try:
         if isinstance(instance.client._client.base_url.host, str):
             provider_url = instance.client._client.base_url.host
+            inference_endpoint = str(instance.client._client.base_url)
```

Comment:
why str is needed ?

Reply:
The base_url is an instance of the URL class, so using str() will providefull URL as a plain string.